lnrpc/routerrpc/router_backend.go

```diff
@@ -648,6 +648,10 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 			payIntent.Amount = *payReq.MilliSat
 		}
 
+		if !payReq.Features.HasFeature(lnwire.MPPOptional) {
```

Comment:
Why's this needed if we'll just fail to split with the changes in `payment_session.go`?

Reply:
It's redudant. I usually opt to root out known bad values at every layer that can. I can take it out if you feel it's not warranted